File: 프로그래머스/lv1/이재표/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer>stack=new Stack<>();
        stack.push(arr[0]);
        for(int i=1;i<arr.length;i++){
            int peek=stack.peek();
            if(peek==arr[i]){
                continue;
            }
            stack.push(arr[i]);
        }
        int[]answer=new int[stack.size()];
        int idx=0;
        for(Integer s:stack){
            answer[idx++]=s;
        }
        return answer;
    }
}

File: 프로그래머스/lv1/leehandsub/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    int now=-1;

    for(int i=0;i<arr.size();i++)
    {
        if(now!=arr[i])
        {
            now=arr[i];
            answer.push_back(now);
        }
        else
            continue;
        
    }
    
    
    return answer;
}
